pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -428,7 +428,7 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 		}
 		p.activeQ.Add(rawPodInfo)
 		metrics.SchedulerQueueIncomingPods.WithLabelValues("active", BackoffComplete).Inc()
-		defer p.cond.Broadcast()
+		p.cond.Broadcast()
```

Comment:
I would change the returns above to breaks, add a flag to indicate whether or not to broadcast and move the broadcast call outside the loop.

Reply:
@ahg-g  Thanks for the tip, updated